streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionWrapperSerde.java

```diff
@@ -43,21 +46,28 @@ public SubscriptionWrapperSerde(final Serde<K> primaryKeySerde) {
     }
 
     private static class SubscriptionWrapperSerializer<K> implements Serializer<SubscriptionWrapper<K>> {
+        private final String primaryKeySerializationPseudoTopic;
         private final Serializer<K> primaryKeySerializer;
-        SubscriptionWrapperSerializer(final Serializer<K> primaryKeySerializer) {
+
+        SubscriptionWrapperSerializer(final String primaryKeySerializationPseudoTopic,
+                                      final Serializer<K> primaryKeySerializer) {
+            this.primaryKeySerializationPseudoTopic = primaryKeySerializationPseudoTopic;
             this.primaryKeySerializer = primaryKeySerializer;
         }
 
         @Override
-        public byte[] serialize(final String topic, final SubscriptionWrapper<K> data) {
+        public byte[] serialize(final String ignored, final SubscriptionWrapper<K> data) {
```

Comment:
Hey @guozhangwang , the actual subscription topic name is defined here: https://github.com/apache/kafka/pull/8054/files#diff-05872eec0e2733128854406f87979013R959
```
final String subscriptionTopicName = renamed.suffixWithOrElseGet("-subscription-registration", builder, SUBSCRIPTION_REGISTRATION) + TOPIC_SUFFIX;
```

The reason we're not using that topic name for the "primary key" (which is actually part of the _value_ here) is what I explained above. I see that my statement wasn't super clear, though...
1. we're already using that pseudo-topic for the primary key in the CombinedKeySchema, so we can avoid proliferating schemas we're registering
2. using the subscription topic would mean that we register the primary key as the "value serde" of the subscription topic, but the values in the topic are actually SubscriptionWrappers, not primary keys. So it would actually be misleading to use the subscription topic for this purpose. At least for the pseudo-topic, it doesn't actually exist in the brokers, so it's not possible to have an inconsistency between the registered serde and the log data.